src/controllers/admin/roles.js

```diff
@@ -0,0 +1,233 @@
+import roleValidate from '../../validation/createRole';
+import roleServices from '../../services/roles';
+import userBadRequest from '../../utils/errorHandling/userBadRequest';
+import applicationError from '../../utils/errorHandling/applicationError';
+import notFound from '../../utils/errorHandling/notFound';
+
+exports.create = async (req, res, next) => {
+    try{
+        /* data validation */
+        const { error } = roleValidate.roleValidation(req.body);
+        if (error) throw new userBadRequest(error.details[0].message);
```

Comment:
validation should happen in middleware not in controller leevel

Reply:
I think so